vaadin-context-menu-flow-parent/vaadin-context-menu-flow-integration-tests/src/test/java/com/vaadin/flow/component/contextmenu/it/MenuItemDisableOnClickIT.java

```diff
@@ -0,0 +1,203 @@
+
+        executeScript(
+                "arguments[0].click();arguments[0].click();arguments[0].click();arguments[0].click();",
+                menuItem);
+        Assert.assertFalse(findElement(By.id(itemId)).isEnabled());
+
+        var disabledWithSingleClickMessage = getEnabledStateChangeMessage(
+                "Disabled on click", false, 1);
+        Assert.assertEquals(disabledWithSingleClickMessage,
+                findElement(By.id(messageId)).getText());
+
+        executeScript("arguments[0].removeAttribute(\"disabled\");"
+                + "arguments[0].click();", findElement(By.id(itemId)));
+        Assert.assertEquals(disabledWithSingleClickMessage,
+                findElement(By.id(messageId)).getText());
+
+        // Test that enabling after disable on click works more than once
+        for (int i = 0; i < 3; i++) {
+            clickElementWithJs("enable-menu-item");
+
+            Assert.assertTrue(findElement(By.id(itemId)).isEnabled());
+            Assert.assertEquals("Re-enabled item from server.",
+                    findElement(By.id(messageId)).getText());
+
+            clickElementWithJs(itemId);
+            Assert.assertFalse(findElement(By.id(itemId)).isEnabled());
+            Assert.assertEquals(disabledWithSingleClickMessage,
+                    findElement(By.id(messageId)).getText());
+        }
+    }
+
+    @Test
+    public void setItemKeepOpenFalse_clickDisableOnClickMenuItem_menuClosed() {
+        var itemId = "disable-on-click-menu-item";
+        clickElementWithJs("toggle-keep-open");
+        rightClickOn(TARGET_ID);
+        verifyOpened();
+        waitUntil(ExpectedConditions.elementToBeClickable(By.id(itemId)), 2);
+        clickElementWithJs(itemId);
+        verifyClosed();
+    }
+
+    @Test
+    public void disableOnClick_enableInSameRoundTrip_clientSideMenuItemIsEnabled() {
+        var itemId = "disable-on-click-re-enable-menu-item";
+        rightClickOn(TARGET_ID);
+        waitUntil(ExpectedConditions
+                .elementToBeClickable(findElement(By.id(itemId))), 2);
+        for (int i = 0; i < 3; i++) {
+            var menuItem = findElement(By.id(itemId));
+            var disabled = (Boolean) getCommandExecutor().getDriver()
+                    .executeAsyncScript("""
+                            var callback = arguments[arguments.length - 1];
+                            var element = arguments[0];
+                            element.click();
+                            requestAnimationFrame(function() {
+                              callback(element.disabled);
+                            });""", menuItem);
+            Assert.assertTrue(disabled);
+            waitUntil(driver -> findElement(By.id(itemId)).isEnabled());
+        }
+    }
+
+    @Test
+    public void removeDisableOnClick_itemWorksNormally() {
+        var itemId = "disable-on-click-menu-item";
+        rightClickOn(TARGET_ID);
+        waitForElementPresent(By.id(itemId));
```

Comment:
**nit:** The button reference could be stored in a variable to avoid selecting it with selectors repeatedly (applies to all tests in this suite)

Reply:
Refactored all the tests to reuse the references.